api/src/main/java/dto/CarDTO.java

```diff
@@ -0,0 +1,94 @@
+package dto;
+
+import java.util.Objects;
+
+/**
+ * Car DTO class for Car entity
+ *
+ * @author Richard Hrmo
+ */
+public class CarDTO {
+
+    private Long id;
+
+    private DriverDTO driver;
+
+    private ComponentDTO engine;
+    private ComponentDTO aerodynamics;
+    private ComponentDTO suspension;
+    private ComponentDTO transmission;
+    private ComponentDTO brakes;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public DriverDTO getDriver() {
+        return driver;
+    }
+
+    public void setDriver(DriverDTO driverDTO) {
+        this.driver = driver;
+    }
+
+    public ComponentDTO getEngine() {
+        return engine;
+    }
+
+    public void setEngine(ComponentDTO engine) {
+        this.engine = engine;
+    }
+
+    public ComponentDTO getAerodynamics() {
+        return aerodynamics;
+    }
+
+    public void setAerodynamics(ComponentDTO aerodynamics) {
```

Comment:
You don't check components type. IMHO if you create our own exception for this it'd be really great. Something like WrongComponentException or so.